sdks/python/apache_beam/transforms/trigger_test.py

```diff
@@ -718,6 +719,9 @@ def fire_timers():
         output = [
             _windowed_value_info(wv)
             for wv in driver.process_elements(state, bundle, watermark)]
+        # ignore WindowValues with expired window in output
+        output = [wvi for wvi in output
+                  if wvi['window'][1] + allowed_lateness >= watermark]
```

Comment:
Above we do filtering before calling driver.process_elements. Here we do filtering after, which seems inconsistent. Any reason allowed_lateness shouldn't be used within `driver.process_elements` (which takes windowing as a parameter) to consistently do the right thing?

Reply:
Seems that the java direct runner and dataflow JRH both filters elements before trigger so I thought it should be better to do it the same way in python. Filter out the elements here is more for reusing of the transcripts in trigger tests since here trigger doesn't really deal with lateness. Another option is to pass the transcript tests that specifies allowed_lateness in trigger tests, would that makes more sense?

Reply:
or we can probably use driver.process_elements to deal with the lateness if we can add input watermark to the function arg list, I'll see if that's possible.